uberenv.py

```diff
@@ -361,16 +375,26 @@ def setup_paths_and_dirs(self):
             if not uberenv_plat is None:
                 self.spack_config_dir = pabs(pjoin(spack_configs_path,uberenv_plat))
 
-        # Spack local packages path
-        spack_packages_path = pabs(pjoin(self.uberenv_path,"packages"))
-        if "spack_packages_path" in self.project_opts.keys():
-            new_path = self.project_opts["spack_packages_path"]
-            if new_path is not None:
-                spack_packages_path = pabs(new_path)
-        # All local packages
-        self.pkgs = pjoin(spack_packages_path, "*")
+        # Find project level packages to override spack's internal packages
+        self.spack_packages_base_paths = []
+        if "spack_packages_base_paths" in self.project_opts.keys():
+            # packages directories listed in project.json
+            _paths = self.project_opts["spack_packages_base_paths"]
+            for _path in _paths:
+                self.spack_packages_base_paths.append(pabs(_path))
+        else:
+            # default to packages living next to uberenv script
+            self.spack_packages_base_paths.append(pabs(pjoin(self.uberenv_path,"packages")))
 
         # setup destination paths
+        if not self.opts["prefix"]:
+            if self.project_opts["force_commandline_prefix"]:
+                # project has specified prefix must be on command line
+                print("[ERROR: --prefix flag for library destination is required]")
+                sys.exit(1)
+            # otherwise set default
+            self.opts["prefix"] = "uberenv_libs"
+            
```

Comment:
I understand the usage.
A general note: options in Uberenv now tend to have diverging treatment. One has a default value, the other is expected to be set in the project.json, another can be forced to be defined on command line, etc.

Reply:
Yes ... we should probably do a once over and conform many things in uberenv.  This was the cleanest way, I felt, to check for a non-user specified path.